### PR TITLE
Refactor AbstractDistributedTransactionManager and AbstractTwoPhaseCommitTransactionManager

### PR DESCRIPTION
**Pull request by brfrn169**
## Description

This PR refactors the `AbstractDistributedTransactionManager` class and the `AbstractTwoPhaseCommitTransactionManager` class.

## Related issues and/or PRs

N/A

## Changes made

- Moved the CRUD operations in the `AbstractDistributedTransactionManager` class and the `AbstractTwoPhaseCommitTransactionManager` class to the implementation classes.

## Checklist

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes.
- [x] Any remaining open issues linked to this PR are documented and up-to-date (Jira, GitHub, etc.).
- [x] Tests (unit, integration, etc.) have been added for the changes.
- [x] My changes generate no new warnings.
- [x] Any dependent changes in other PRs have been merged and published.

## Additional notes (optional)

N/A

## Release notes

N/A

### CODE REVIEW

File: core/src/main/java/com/scalar/db/common/AbstractDistributedTransactionManager.java [hunk 1]
```diff
@@ -76,130 +61,6 @@ public Optional<String> getTable() {
     return tableName;
   }
 
-  @Override
-  public Optional<Result> get(Get get) throws CrudException, UnknownTransactionStatusException {
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/common/AbstractDistributedTransactionManager.java:80]: Moved the CRUD operations to the implementation classes.

File: core/src/main/java/com/scalar/db/common/AbstractTwoPhaseCommitTransactionManager.java [hunk 2]
```diff
@@ -78,134 +61,6 @@ public Optional<String> getTable() {
     return tableName;
   }
 
-  @Override
-  public Optional<Result> get(Get get) throws CrudException, UnknownTransactionStatusException {
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/common/AbstractTwoPhaseCommitTransactionManager.java:82]: Diito. Moved the CRUD operations to the implementation classes.

File: core/src/main/java/com/scalar/db/transaction/consensuscommit/ConsensusCommitManager.java [hunk 3]
```diff
@@ -209,7 +231,137 @@ DistributedTransaction begin(String txId, Isolation isolation, SerializableStrat
...
+    } catch (TransactionException e) {
+      rollbackTransaction(transaction);
+      throw new CrudException(e.getMessage(), e, e.getTransactionId().orElse(null));
+    }
+  }
+
+  @VisibleForTesting
+  DistributedTransaction beginInternal() throws TransactionException {
+    String txId = UUID.randomUUID().toString();
+    return begin(txId, config.getIsolation(), config.getSerializableStrategy(), false);
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/transaction/consensuscommit/ConsensusCommitManager.java:356]: We don't need to decorate the transaction object for the one-shot CRUD operations.

File: core/src/main/java/com/scalar/db/transaction/consensuscommit/TwoPhaseConsensusCommitManager.java [hunk 4]
```diff
@@ -162,7 +186,142 @@ private TwoPhaseCommitTransaction createNewTransaction(
...
+      rollbackTransaction(transaction);
+      throw new CrudException(e.getMessage(), e, e.getTransactionId().orElse(null));
+    }
+  }
+
+  @VisibleForTesting
+  TwoPhaseCommitTransaction beginInternal() throws TransactionException {
+    String txId = UUID.randomUUID().toString();
+    return createNewTransaction(
+        txId, config.getIsolation(), config.getSerializableStrategy(), false);
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/transaction/consensuscommit/TwoPhaseConsensusCommitManager.java:316]: Ditto. We don't need to decorate the transaction object for the one-shot CRUD operations.

File: core/src/main/java/com/scalar/db/transaction/jdbc/JdbcTransactionManager.java [hunk 5]
```diff
@@ -142,6 +164,136 @@ public DistributedTransaction start(
...
+    } catch (TransactionException e) {
+      rollbackTransaction(transaction);
+      throw new CrudException(e.getMessage(), e, e.getTransactionId().orElse(null));
+    }
+  }
+
+  @VisibleForTesting
+  DistributedTransaction beginInternal() throws TransactionException {
+    String txId = UUID.randomUUID().toString();
+    return begin(txId, false);
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/transaction/jdbc/JdbcTransactionManager.java:286]: Ditto. We don't need to decorate the transaction object for the one-shot CRUD operations.

**Review by komamitsu — APPROVED** [review]
LGTM! 👍

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Review by Torch3333 — APPROVED** [review]
LGTM, thank you!

**Pull request merged**